pkg/pool-hooks/contracts/hooks-quantamm/HyperSurgeHook.sol

```diff
@@ -0,0 +1,747 @@
+    /// @param tokenIndex The index of the token to configure (0..7).
+    /// @param pairIdx the index of the pair being set
+    function setTokenPriceConfigIndex(
+        address pool,
+        uint8 tokenIndex,
+        uint32 pairIdx
+    ) external onlySwapFeeManagerOrGovernance(pool) {
+        //TODO should this be done on construction? Not sure there is any reason to change it
+        //or at least be blocked once set
+        TokenPriceCfg memory tempCfg;
+        PoolDetails memory details = _poolCfg[pool].details;
+
+        if (!details.initialized) {
+            revert PoolNotInitialized();
+        }
+        if (tokenIndex >= details.numTokens) {
+            revert TokenIndexOutOfRange();
+        }
+
+        if (pairIdx == 0) {
+            revert InvalidPairIndex();
+        }
+
+        uint8 sz = HyperTokenInfo.szDecimals(pairIdx); // may revert "dec"
+
+        if (sz > 6) {
+            revert InvalidDecimals();
+        }
+
+        tempCfg.pairIndex = pairIdx;
+        tempCfg.priceDivisor = _divisorFromSz(sz); // precompute to avoid EXP in hot path
+
+        _poolCfg[pool].tokenCfg[tokenIndex] = tempCfg;
+
+        emit TokenPriceConfiguredIndex(pool, tokenIndex, tempCfg.pairIndex, sz);
+    }
+
+    struct SetBatchConfigs {
+        uint8 sz;
+        TokenPriceCfg tempCfg;
+        uint256 i;
+    }
+
+    /// @notice Batch version (indices).
+    /// @param pool the pool address
+    /// @param tokenIndices the indices of the token configs being changed
+    /// @param pairIdx the index of the pair being changed
+    function setTokenPriceConfigBatchIndex(
+        address pool,
+        uint8[] calldata tokenIndices,
+        uint32[] calldata pairIdx
+    ) external onlySwapFeeManagerOrGovernance(pool) {
+        //TODO should this be done on construction? Not sure there is any reason to change it
+        //or at least be blocked once set
+        PoolDetails memory detail = _poolCfg[pool].details;
+        SetBatchConfigs memory cfg;
+
+        if (!detail.initialized) {
+            revert PoolNotInitialized();
+        }
+
+        if (tokenIndices.length != pairIdx.length) {
+            revert InvalidArrayLengths();
+        }
+
+        for (cfg.i = 0; cfg.i < tokenIndices.length; ++cfg.i) {
+            if (tokenIndices[cfg.i] >= detail.numTokens) {
+                revert TokenIndexOutOfRange();
+            }
+
+            if (pairIdx[cfg.i] == 0) {
+                revert InvalidPairIndex();
+            }
+
+            cfg.sz = HyperTokenInfo.szDecimals(pairIdx[cfg.i]); // may revert "dec"
+
+            if (cfg.sz > 6) {
+                revert InvalidDecimals();
+            }
+
+            cfg.tempCfg.pairIndex = pairIdx[cfg.i];
+            cfg.tempCfg.priceDivisor = _divisorFromSz(cfg.sz);
+
+            _poolCfg[pool].tokenCfg[tokenIndices[cfg.i]] = cfg.tempCfg;
+
+            emit TokenPriceConfiguredIndex(pool, tokenIndices[cfg.i], cfg.tempCfg.pairIndex, cfg.sz);
```

Comment:
This code and the code in `setTokenPriceConfigIndex` are exactly the same, maybe this function could call that one and avoid code duplicity (with probably an extra private function to avoid accessing PoolDetails twice)?